src/api/search.ts

```diff
@@ -116,40 +124,91 @@ const getProductSearch = async ({
     sort
   );
 
-  window.adobeDataLayer.push((dl : any) => {
-    dl.push({ event: 'search-request-sent', eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID } })
-  })
+  window.adobeDataLayer.push((dl: any) => {
+    dl.push({
+      event: 'search-request-sent',
+      eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID },
+    });
+  });
   // ======  end of data collection =====
 
-  const query = categorySearch && categoryId ? CATEGORY_QUERY : PRODUCT_SEARCH_QUERY;
+  const query =
+    categorySearch && categoryId ? CATEGORY_QUERY : PRODUCT_SEARCH_QUERY;
+
   const results = await fetch(apiUrl, {
     method: 'POST',
     headers,
     body: JSON.stringify({
-      query: query.replace(/(?:\r\n|\r|\n|\t|[\s]{4})/g, ' ').replace(/\s\s+/g, ' '),
+      query: query
+        .replace(/(?:\r\n|\r|\n|\t|[\s]{4})/g, ' ')
+        .replace(/\s\s+/g, ' '),
       variables: { ...variables },
     }),
   }).then((res) => res.json());
 
+  const items = results?.data?.productSearch?.items ?? [];
+
+  const productIds = items.map((item: Product) => item.product.id);
+
+  const productLabelsResults = await getGraphQL(GET_PRODUCT_LABELS_QUERY, {
```

Comment:
can we run this query in parallel instead of sequential calls ?

Reply:
i looked into this but im not sure if this is possible.  perhaps we can sync tomorrow to see what we can do.  i explored looking at `Promise.all` but im not quite sure this would satisfy

Reply:
@yugandhar02 i made an attempt. please see if this is what you meant

Reply:
@johnnyung91 I overlooked the fact that this query needs product ids from the search result
So it needs to be sequential only. 
The issue is this will delay the page render
One way to solve is to use separate state in products.tsx and fetch labels once search results are available but that will trigger another render cycle 
For now, we can keep it as it is